orte/mca/odls/default/odls_default_module.c

```diff
@@ -455,9 +455,16 @@ static int do_child(orte_odls_spawn_caddy_t *cd, int write_fd)
     /* Exec the new executable */
     execve(cd->cmd, cd->argv, cd->env);
     getcwd(dir, sizeof(dir));
+    struct stat stats;
+    char* msg;
+    if(0 == stat(cd->app->app, &stats)) {
```

Comment:
What if the app is a binary with unresolved dependencies?
I am afraid the error message will be "Bad interpreter" in that case.

Reply:
I agree; this would batch too many failure modes (wrong architecture, missing libraries, wrong os, etc.) into one error code, and worse, it would be coming from us, not the OS>

Reply:
one option could be to `stat()` the file, and if successful and it starts with `#!`, then add a hint into the error message that the MPI app is indeed a script and the error could come from the interpreter.

Reply:
On Linux, it looks like `errno == ENOENT` means either the file was missing or it's a bad interpreter.  Looks like it's the same behavior on Linux.

So adding a check for `errno == ENOENT` would likely be sufficient for this small fix.